### ISSUE TITLE
Remove kernel from /boot when the  booster-remove script is called 

### ISSUE THREAD

**Issue by tam1m** [anatol/booster#102]
I think booster should remove the `/boot/vmlinuz-$kernel` when the remove hook is executed. Most kernel packages don't install the kernel to `/boot` themselves and instead, let the initramfs generators do the job.
 
Mkinitpico does this, too (see https://github.com/archlinux/mkinitcpio/blob/master/libalpm/scripts/mkinitcpio-remove#L36)

**Comment by anatol** [comment]
I believe there was an activity to consolidate kernel management and make it generic for all the Arch's initamfs generators. СС @grazzolini for more comments on it.

**Comment by tam1m** [comment]
Yes, there is this https://github.com/archlinux/mkinitcpio/pull/7 (see the discussions and https://github.com/archlinux/mkinitcpio/commit/9c1e0f3dd01151b79e9b557ea8510605616c1369).

Basically, mkinitcpio only removes vmlinuz from /boot, if `lib/modules/$kernel/pkgbase` exists and otherwise bails. In those cases the responsibility is left to the kernel package.

**Comment by anatol** [comment]
We try to make multi-initramfs configuration possible. i.e. if both booster and initramfs are installed then booster should not delete files used by initramfs.

I think installing vmlinuz should really be done by `linux` package itself. initramfs should be responsible for image generation/removal only.

**Comment by grazzolini** [comment]
@anatol I'm all up for having generic hooks that copies vmlinuz to /boot and also handle initramfs generation, of any of the generators we have on the repository. And I started doing some work in that direction. However, the more I did that, the more I looked at kernel-install and the more using something like it made sense.

The reason for not having the kernel package copy vmlinuz to /boot anymore is that you can have setups that don't need /boot, many people today have to use their ESP as /boot or juggle files around.

**Comment by oech3** [comment]
Managing vmlinuz in ESP by initramfs generator side is really bad while Arch allows booting without initramfs
(https://wiki.archlinux.org/title/Arch_boot_process#Running_without_initramfs).
We lose vmlinuz if we uninstall booster,mkinitcpio and dracut.
It should be managed by kernel package. 
(I cant open issue for some reason.)

**Comment by oech3** [comment]
https://gitlab.archlinux.org/archlinux/packaging/packages/linux/-/issues/125
It seems kernel package don't want to have hooks to manage `vmlinuz-*`.

We can mitigate competition of initramfs packages and complexity of copy/remove hook if `mkinitcpio,dracut,` and `booster` share a standalone hook package to manage `vmlinuz`.
This approach avoid to include hook for each kernel package.

**Comment by oech3** [comment]
Current kernel package does'n t allow to remove every initramfs gegerator.
This approact still fail if it is allowed.

**Comment by oech3** [comment]
Is this issue solved by importing https://aur.archlinux.org/packages/pacman-hook-kernel-install to Extra repository?
